airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialReadUtil.java

```diff
@@ -218,23 +234,31 @@ public static List<AutoCloseableIterator<AirbyteMessage>> getCdcReadIterators(fi
           getMySqlInitialLoadHandler(database, emittedAt, quoteString, initialLoadStreams, initialLoadGlobalStateManager,
               Optional.of(new CdcMetadataInjector(emittedAt.toString(), stateAttributes, metadataInjector)));
 
-      // Because initial load streams will be followed by cdc read of those stream, we only decorate with
-      // complete status trace
-      // after CDC read is done.
+      // Start and complete stream status messages are emitted while constructing the full set of initial
+      // load and incremental debezium iterators.
       initialLoadIterator.addAll(initialLoadHandler.getIncrementalIterators(
           new ConfiguredAirbyteCatalog().withStreams(initialLoadStreams.streamsForInitialLoad()),
           tableNameToTable,
-          emittedAt, true, false));
+          emittedAt, false, false, Optional.of(initialLoadTimeout)));
     }
 
+    // CDC stream status messages should be emitted for streams.
     final List<AutoCloseableIterator<AirbyteMessage>> cdcStreamsStartStatusEmitters = catalog.getStreams().stream()
-        .filter(stream -> !initialLoadStreams.streamsForInitialLoad.contains(stream))
+        .filter(stream -> stream.getSyncMode() == SyncMode.INCREMENTAL)
         .map(stream -> (AutoCloseableIterator<AirbyteMessage>) new StreamStatusTraceEmitterIterator(
             new AirbyteStreamStatusHolder(
                 new io.airbyte.protocol.models.AirbyteStreamNameNamespacePair(stream.getStream().getName(), stream.getStream().getNamespace()),
                 AirbyteStreamStatusTraceMessage.AirbyteStreamStatus.STARTED)))
         .toList();
 
+    final List<AutoCloseableIterator<AirbyteMessage>> cdcStreamsEndStatusEmitters = catalog.getStreams().stream()
+        .filter(stream -> stream.getSyncMode() == SyncMode.INCREMENTAL)
+        .map(stream -> (AutoCloseableIterator<AirbyteMessage>) new StreamStatusTraceEmitterIterator(
+            new AirbyteStreamStatusHolder(
+                new io.airbyte.protocol.models.AirbyteStreamNameNamespacePair(stream.getStream().getName(), stream.getStream().getNamespace()),
```

Comment:
let's import this? `io.airbyte.protocol.models.AirbyteStreamNameNamespacePair`

Reply:
Unfortunately, there are two flavors of `AirbyteStreamNameNamespacePair` v0 vs otherwise. Here, we need to explicitly include this to avoid collision iwth the other object